store/streaming/abci/examples/stdout/stdout.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-plugin"
+	abci "github.com/tendermint/tendermint/abci/types"
+
+	streamingabci "cosmossdk.io/store/streaming/abci"
+	store "cosmossdk.io/store/types"
+)
+
+// StdoutPlugin is the implementation of the ABCIListener interface
+// For Go plugins this is all that is required to process data sent over gRPC.
+type StdoutPlugin struct {
+	BlockHeight int64
+}
+
+func (a StdoutPlugin) ListenBeginBlock(ctx context.Context, req abci.RequestBeginBlock, res abci.ResponseBeginBlock) error {
+	a.BlockHeight = req.Header.Height
+	// process tx messages (i.e: sent to external system)
+	fmt.Printf("listen-begin-block: block-height=%d req=%v res=%v", a.BlockHeight, req, res)
+	return nil
+}
```

Comment:
This method is defined on a value receiver, so setting a.BlockHeight will only apply to this function call, it won't be visible to other consumers.

Reply:
fixed.